spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/ReadIntegrationTestBase.java

```diff
@@ -375,4 +379,112 @@ public void testArrowCompressionCodec() {
 
     assertThat(avroResults).isEqualTo(arrowResultsForLZ4FrameCodec);
   }
+
+  private void uploadFileToGCS(String resourceName, String destinationURI, String contentType) {
+    try {
+      AcceptanceTestUtils.uploadToGcs(
+          getClass().getResourceAsStream("/integration/" + resourceName),
+          destinationURI,
+          contentType);
+    } catch (Exception exception) {
+      exception.printStackTrace();
+    }
+  }
+
+  @Test
+  public void testReadFromBigLakeTable_csv() {
+    String temporaryGCSURI =
+        String.format(
+            "gs://%s/%s", TestConstants.temporaryGcsBucket, TestConstants.SHAKESPEARE_CSV_FILENAME);
+    String table = testTable + "_csv";
+    uploadFileToGCS(TestConstants.SHAKESPEARE_CSV_FILENAME, temporaryGCSURI, "text/csv");
+    IntegrationTestUtils.createExternalTable(
+        testDataset.toString(),
+        table,
+        TestConstants.SHAKESPEARE_TABLE_SCHEMA,
+        temporaryGCSURI,
+        FormatOptions.csv());
+    Dataset<Row> df =
+        spark
+            .read()
+            .format("bigquery")
+            .option("dataset", testDataset.toString())
+            .option("table", table)
+            .load();
+    testShakespeare(df);
+  }
+
+  @Test
+  public void testReadFromBigLakeTable_json() {
+    String temporaryGCSURI =
+        String.format(
+            "gs://%s/%s",
+            TestConstants.temporaryGcsBucket, TestConstants.SHAKESPEARE_JSON_FILENAME);
+    String table = testTable + "_json";
+    uploadFileToGCS(TestConstants.SHAKESPEARE_JSON_FILENAME, temporaryGCSURI, "application/json");
+    IntegrationTestUtils.createExternalTable(
+        testDataset.toString(),
+        table,
+        TestConstants.SHAKESPEARE_TABLE_SCHEMA,
+        temporaryGCSURI,
+        FormatOptions.json());
+    Dataset<Row> df =
+        spark
+            .read()
+            .format("bigquery")
+            .option("dataset", testDataset.toString())
+            .option("table", table)
+            .load();
+    testShakespeare(df);
+  }
+
+  @Test
+  public void testReadFromBigLakeTable_avro() {
+    String temporaryGCSURI =
```

Comment:
this method should have assertThrows(), as avro is not supported yet

Reply:
Addressed